tests/filecheck/dialects/riscv/riscv_assembly_emission.mlir

```diff
@@ -174,6 +174,21 @@
     // CHECK-NEXT:   hello j3, j4, zero, j1
 
 
+    // RISC-V Extensions
+
+    riscv.frep_outer %0, 0, 0 ({
+      %add_o = riscv.add %0, %1 : (!riscv.reg<zero>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    }) : (!riscv.reg<zero>) -> ()
+
+    // CHECK:          frep.outer zero, 1, 0, 0
+    // CHECK-NEXT:     add  j2, zero, j1
```

Comment:
I don't think this is valid, because the `xfrep` is for floating point instructions only IIUC.
The FPU and the integer ALU are fully decoupled in snitch, and there's only a FPU sequence buffer, no Integer instruction sequence buffer IIUC.
Maybe we should make sure non-floating point instructions don't pass verification?

Reply:
That escaped me as well, the "f" in `frep` stands for floating-point as I've seen it in the paper and I think Verilator complained when I had tried an int instr

https://github.com/pulp-platform/snitch/blob/d026f47843f0ea6c269244c4e6851e0e09141ec3/hw/ip/snitch/src/snitch.sv#L1972C25-L1972C25

Reply:
!! I thought it was "Fixed". We don't currently have a way to check for fp instructions generally, we'll have to make it a set of classes. In any case I'd propose to do this in a downstream PR if that's ok?

Reply:
Sure @superlopuh !